### PR TITLE
dbeaver/dbeaver#34042 Explain execution plan parsing error when TRCLOG_DETAIL_MTRNODE = 1 option used

### LINKED ISSUE #34042: Altibase: Explain execution plan parsing error when TRCLOG_DETAIL_MTRNODE = 1 option used.
### Description

Parsing error occurs at explain execution plan for Altibase when the following query has been set to print MtrNode.
ALTER SYSTEM SET TRCLOG_DETAIL_MTRNODE = 1;

### DBeaver Version

Community edition 24.0.0

### Operating System

Windows 11

### Database and driver

Altibase 7.3.0 or higher

### Steps to reproduce

1. Run the following SQL statements
DROP TABLE T50764;
CREATE TABLE T50764 ( I1 INT, I2 INT, I3 INT );
INSERT INTO T50764 VALUES ( 1,1,1);
INSERT INTO T50764 VALUES ( 2, 2, 1);
INSERT INTO T50764 VALUES ( 3,1, 1);
ALTER SYSTEM SET TRCLOG_DETAIL_MTRNODE = 1;

2. Click 'Explain execution plan' with the following SQL at DBeaver SQL editor -> Parsing error
SELECT * FROM T50764 WHERE I1 > 0 AND I2 > 0 ORDER BY I1; -> Explain execution plan

image

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zennken**
Hello,

I am happy to ask a pull-request for https://github.com/dbeaver/dbeaver/issues/34042

New code was checked with Check Style to ensure compliance.

Thanks,

### CODE REVIEW

**Comment by zennken** [comment]
Hello,  @LonwoLonwo

There hasn't been any progress for a month, so I'm leaving this comment in case you've forgotten. It's okay if you plan to do it later. :) 

Thanks,

**Comment by E1izabeth** [comment]
Hi, @zennken 
Sorry for long waiting. We missed your PR. Will try to review shortly

**Comment by zennken** [comment]
> Hi, @zennken Sorry for long waiting. We missed your PR. Will try to review shortly

I am happy to receive your reply.  It's not urgent one, so take your time :)

**Review by E1izabeth — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/plan/AltibasePlanBuilder.java [hunk 1]
```diff
@@ -50,87 +58,218 @@ public static List<AltibasePlanNode> build(AltibaseDataSource dataSource, String
...
-                + "       SCAN ( TABLE: ORDERS ORDERS_03, INDEX: ODR_IDX2, RANGE SCAN, ACCESS: ??, COST: 0.41 ) " + "\n"
-                + "        [ VARIABLE KEY ]" + "\n"
-                + "        OR" + "\n"
-                + "         AND" + "\n"
-                + "          CUSTOMERS.CNO = ORDERS_03.CNO" + "\n"
-                + "        [ FILTER ]" + "\n"
-                + "        ORDERS_03.QTY >= 10000" + "\n"
-                + "-----------------------------------------------------------"
-                + "* SIMPLE QUERY PLAN";
+        String plan =
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/plan/AltibasePlanBuilder.java:141]: Tests are cool, but can we convert them to actual tests?

Tests are located [here](https://github.com/dbeaver/dbeaver/tree/devel/test). You can create a new bundle and move tests there.

Comment (zennken) [reply plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/plan/AltibasePlanBuilder.java:141]: Hello,  @ShadelessFox

Thanks for your comment. I will implement the test according to your advice.

Have a great day :)

Comment (zennken) [reply plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/plan/AltibasePlanBuilder.java:141]: Hi @ShadelessFox,

Could you let me know what kind of archetype should be used to create a test Maven module?

tests/pom.xml -> New Maven Module Project -> org.jkiss.dbeaver.ext.altibase.test -> **_Select an Archetype_** ?

Any reference or help will be appreciated :)

====

I tried to create a new Maven module with "Create a simple project (skip archetype selection" option, the test case is working but pom.xml shows an error as follows.

image

Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/plan/AltibasePlanBuilder.java:141]: The archetype is not required.

I don't use Eclipse for development, so I'm unsure why this error occurs. Have you tried building DBeaver from the command line?

Comment (zennken) [reply plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/plan/AltibasePlanBuilder.java:141]: Thanks for your reply :)

Building DBeaver with build.sh at command line looks working correctly as well as my new test. The problem may comes from Eclipse or maven cache. I'll try to find the root cause.

image

image

Comment (zennken) [reply plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/plan/AltibasePlanBuilder.java:141]: I tried to figure out the the error at pom.xml, but I couldn't. (https://stackoverflow.com/questions/65677156/could-not-transfer-artifact-from-to-central-intellij) But, what I know is that it comes from not my code but m2 repository cache.

When I tried execute 'mvn clean verify' at dbeaver/test or build.sh at dbeaver at command line, everything is fine. So, I committed my new test bundle. 

If any problem found, please let me know.

Thanks,

Error message at pom.xml:
```
org.slf4j:slf4j-api:jar:1.7.36 failed to transfer from https://repo.maven.apache.org/maven2 during a previous attempt. 
This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced. 
Original error: Could not transfer artifact org.slf4j:slf4j-api:jar:1.7.36 from/to central (https://repo.maven.apache.org/maven2): The operation was cancelled.
```

**Review by E1izabeth — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/plan/AltibasePlanBuilder.java (hunk 1 above)
Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/plan/AltibasePlanBuilder.java:141]: Thank you

Comment (zennken) [reply plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/plan/AltibasePlanBuilder.java:141]: > Thank you

I should say thank you for your help :)

**Review by ShadelessFox — APPROVED** [review]

**Comment by E1izabeth** [comment]
We passed this PR to our QA team for testing

**Comment by Matvey16** [comment]
Verified

**Pull request merged**